buildSrc/src/test/java/org/elasticsearch/gradle/BuildExamplePluginsIT.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.gradle;
+
+import org.apache.commons.io.FileUtils;
+import org.elasticsearch.gradle.test.GradleIntegrationTestCase;
+import org.gradle.testkit.runner.GradleRunner;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.StandardOpenOption;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class BuildExamplePluginsIT extends GradleIntegrationTestCase {
+
+    private static List<File> EXTERNAL_PROJECTS = Arrays.stream(
+        Objects.requireNonNull(System.getProperty("test.build-tools.plugin.examples"))
+            .split(File.pathSeparator)
+    ).map(File::new).collect(Collectors.toList());
+
+    @Rule
+    public TemporaryFolder tmpDir = new TemporaryFolder();
+
+    @BeforeClass
+    public static void assertProjectsExist() {
+        List<File> existing = EXTERNAL_PROJECTS.stream().filter(File::exists).collect(Collectors.toList());
+        assertEquals(EXTERNAL_PROJECTS, existing);
+    }
+
+    @AfterClass
+    public static void assertEverythingTested() {
+        assertEquals(
+            "Some example plugins are not tested: " + EXTERNAL_PROJECTS,
+            0, EXTERNAL_PROJECTS.size()
+        );
+    }
+
+    public void testCustomSettings() throws IOException {
+        runAndRemove("custom-settings");
+    }
+
+    public void testPainlessWhitelist() throws IOException {
+        runAndRemove("painless-whitelist");
+    }
+
+    public void testRescore() throws IOException {
+        runAndRemove("rescore");
+    }
+
+    public void testRestHandler() throws IOException {
```

Comment:
I don't like these being hardcoded. We will definitely forget to add new examples here.

Reply:
The reason they are hard coded is that we still run all even if one fails, without rewriting this logic in unit + muting tests is easier too.  

We will not  forget to add new ones as in this case the test will fail.
Look at `@AfterClass` i'm not happy about how it changes that static state of the test class, but since it's a test I think it's ok. Essentially we pop the plugin from the list at the start of each test and assert that each plugin has one.

I initially had a loop and changed it to this as soon as I realized we might wan to take decisions based on individual plugins.